pkg/issuer/acme/http/pod.go

```diff
@@ -131,13 +132,27 @@ func (s *Solver) cleanupPods(ctx context.Context, ch *v1alpha1.Challenge) error
 // createPod will create a challenge solving pod for the given certificate,
 // domain, token and key.
 func (s *Solver) createPod(ch *v1alpha1.Challenge) (*corev1.Pod, error) {
-	return s.Client.CoreV1().Pods(ch.Namespace).Create(s.buildPod(ch))
+	return s.Client.CoreV1().Pods(ch.Namespace).Create(
+		s.buildPod(ch))
 }
 
 // buildPod will build a challenge solving pod for the given certificate,
 // domain, token and key. It will not create it in the API server
 func (s *Solver) buildPod(ch *v1alpha1.Challenge) *corev1.Pod {
+	pod := s.buildDefaultPod(ch)
+
+	// Override defaults if they have changed in the pod template.
+	if ch.Spec.Solver != nil {
+		pod = s.mergePodObjectMetaWithPodTemplate(pod,
+			ch.Spec.Solver.HTTP01.Ingress.PodTemplate)
```

Comment:
I think we need to extend this to also check for nil in `ch.Spec.Solver.HTTP01` and `ch.Spec.Solver.HTTP01.Ingress`?

Also, `ch.Spec.Solver` should pretty much always be set (although do continue to check it if it's a pointer 😅)